src/EnergyPlus/WindowManager.cc

```diff
@@ -7945,13 +7943,16 @@ namespace WindowManager {
 					//     Proceed in reverse order such that the last calculation yields zero sun angle to window screen normal (angles=0,0).
 					//     The properties calculated at zero sun angle are then used elsewhere prior to the start of the actual simulation.
 					for ( j = N; j >= 1; --j ) {
+						Real64 SunAzimuth = ( 90.0 / N ) * ( j - 1 ) * ( Pi / 180.0 ); // Azimuth angle of sun during integration
+						Real64 sin_SunAzimuth = std::sin( SunAzimuth );
+						Real64 cos_SunAzimuth = std::cos( SunAzimuth );
 						for ( i = M; i >= 1; --i ) {
-							SunAzimuth = ( 90.0 / N ) * ( j - 1 ) * ( Pi / 180.0 );
-							SunAltitude = ( 90.0 / M ) * ( i - 1 ) * ( Pi / 180.0 );
-							SkyArea = std::sin( SunAltitude ) * std::cos( SunAltitude );
-							//         Integrate transmittance using coordiante transform
-							RelativeAzimuth = std::asin( std::sin( SunAltitude ) * std::cos( SunAzimuth ) ); // phi prime
-							RelativeAltitude = std::atan( std::tan( SunAltitude ) * std::sin( SunAzimuth ) ); // alpha
+							Real64 SunAltitude = ( 90.0 / M ) * ( i - 1 ) * ( Pi / 180.0 ); // Altitude angle of sun during integration
```

Comment:
SunAzimuth, SunAltitude, and SkyArea could all be precalculated outside of the SurfNum loop since N and M are constants. So could RelativeAzimuth and RelativeAltitude